### PR TITLE
Bittrex ws

### PR DESCRIPTION
**Pull request by adrianbegi**
Implemented bittrex websocket
https://bittrex.github.io/api/v3#definition-SocketResponse

Updated from v1 implementation to v3 implementation
https://bittrex.github.io/guides/v3/upgrade

Implemented OnGetPositionWebSocket although Bittrex only returns orders not positions.

### CODE REVIEW

File: src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI_WebSocket.cs [hunk 1]
```diff
@@ -27,253 +27,262 @@ MIT LICENSE
...
 #if HAS_SIGNALR
+		const string URL = "https://socket-v3.bittrex.com/signalr";
 
-		/// <summary>
-		/// Ideally this would be one instance per exchange instance, but Bittrex simply does not work if you request multiple end points over a single hub connection, sigh...
-		/// </summary>
-		public sealed class BittrexWebSocketManager : SignalrManager
+		public string[] channels = new string[] {
+				"heartbeat",
+				"trade_BTC-USD",
```
Comment (EmergentCybernetics) [inline src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI_WebSocket.cs:43]: @adrianbegi What's the reason for `trade_BTC-USD` here? I'm not following what this channel is used for.

File: src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI_WebSocket.cs [hunk 2]
```diff
@@ -27,253 +27,262 @@ MIT LICENSE
...
+			catch (Exception ex)
+			{
+
+				throw ex;
 			}
-			return await new BittrexWebSocketManager().SubscribeToSummaryDeltasAsync(innerCallback, marketSymbols);
+
 		}
 
-		protected override async Task<IWebSocket> OnGetDeltaOrderBookWebSocketAsync
```
Comment (EmergentCybernetics) [inline src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI_WebSocket.cs:165]: @adrianbegi We'll need you to implement the `OnGetDeltaOrderBookWebSocketAsync()` function for Bittrex v3 WS next.

File: src/ExchangeSharp/Utility/SignalrManager.cs [hunk 3]
```diff
@@ -37,6 +40,7 @@ namespace ExchangeSharp
     /// </summary>
     public class SignalrManager
```
Comment (EmergentCybernetics) [inline src/ExchangeSharp/Utility/SignalrManager.cs:41]: @adrianbegi It looks like you ended up not using `SignalrManager`. Let's go ahead and remove the file as it's no longer needed.

**Review by EmergentCybernetics — CHANGES_REQUESTED** [review]
@adrianbegi First pass here looks good. The order/position updates are working for me now. We do need to get `OnGetDeltaOrderBookWebSocketAsync()` added next. It only needs to support the public WebSocket. Please take a look at that when you can.

Thanks!

File: src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI_WebSocket.cs (hunk 1 above)
Comment (adrianbegi) [reply src/ExchangeSharp/API/Exchanges/Bittrex/ExchangeBittrexAPI_WebSocket.cs:43]: Sorry I just added that for testing to make sure I can subscrive to trades, will remove

**Review by EmergentCybernetics — APPROVED** [review]

**Comment by EmergentCybernetics** [comment]
Closing. Merged via https://github.com/EmergentCybernetics/ExchangeSharp/pull/2

**Pull request closed without merging**